### ISSUE TITLE
PHP Fatal error:  Allowed memory size

### ISSUE THREAD

**Issue by atasingorneq** [friendica/friendica#6395]
### Expected behavior
No error

### Actual behavior
Since one month on every worker call this error appears:
`PHP Fatal error:  Allowed memory size of 188743680 bytes exhausted (tried to allocate 67112960 bytes) in /src/Util/JsonLD.php on line 109`

### Friendica version you encountered the problem
2019.01-rc - 1292

### Friendica source (git, zip)
git

### PHP version
7.2.13
`memory_limit = 350M`
The error also appears, if i rise the memory_limit up to 500M

### SQL version
5.5.5-10.0.34-MariaDB

**Comment by MrPetovan** [comment]
Thanks for the report, it is a known occurrence but without the context (full trace) it is impossible to fix as JsonLD is used in all ActivityPub communication, and it doesn't fail consistently.

**Comment by atasingorneq** [comment]
What information you need? Shall i look for some special keywords in my logfile?

**Comment by MrPetovan** [comment]
Do you have any relevant log lines below the one you pasted in this issue? It could help determining what kind of ActivityPub call makes JsonLD loop.

**Comment by atasingorneq** [comment]
No, there are no more log lines. This is an announcement from the cron log.
Today I set the php log level to _report all_ and the friendica log to _Debug_. Than i remove the worker from the crontab and start the worker manuell, but i did not find any information in the logs. If you have an idea for a relevant keyword in the logfiles, you suggestion are welcome ;-)

**Comment by MrPetovan** [comment]
I was expecting a full stack trace with the Fatal Error message, but maybe it is only available with XDebug installed, which we don't recommend for production use. Thanks for the elaboration though.

**Comment by mark4409** [comment]
I can confirm that I'm also getting very regular entries like this, but unfortunately no stack trace. Also on PHP 7.2

**Comment by MrPetovan** [comment]
Ping @annando

**Comment by annando** [comment]
Hmm, there's something wrong with this profile.

**Comment by gitkald** [comment]
I deleted some contacts. After this, I get the same error. I tried to insert ini_set('memory_limit','512M'); in the php file, but no effect. Same error.

Using: Friendica 'The Tazmans Flax-lily' 2019.01 - 1293, Master branch.

**Comment by annando** [comment]
I have some test code. It would be great if someone could test it, before we think about a better solution.

Please edit the file ```vendor/friendica/json-ld/jsonld.php``` (and save the original one).

Go to line 5747 (This is the function ```_parse_json```). There please add these lines:

```
                if (($error != JSON_ERROR_NONE) && empty($rval)) {
                        $error = JSON_ERROR_NONE;
                        $rval = new stdClass();
                }
```

Please add it before the line ```switch ($error)```.

If this does work, then we have to see how to make a final solution.

**Comment by atasingorneq** [comment]
OK, i will test it.

**Comment by atasingorneq** [comment]
I apply the patch, wait one hour and no errors appears. I think, it works.

**Comment by rabuzarus** [comment]
@annando 
Unfortunately, your test code doesn't resolve the fatal error on my system :-/

**Comment by atasingorneq** [comment]
The same behavior here. At first there were fewer errors, but now there are more than before.

**Comment by annando** [comment]
Could you post the exact message, so that we can check whether it is the same line or some other place?

**Comment by atasingorneq** [comment]
It is the same error
`PHP Fatal error:  Allowed memory size of 188743680 bytes exhausted (tried to allocate 48238592 bytes) in /var/www/virtual/paddler/html/src/Util/JsonLD.php on line 118`

**Comment by annando** [comment]
Okay.

**Comment by gitkald** [comment]
I have the same error on my system. After one week, I got this error. If the error comes up, I ran the crontab worker command manualy and this fix it for one week.

Normaly I thought, that command line and cron are the same. But I saw, that the cronjob runs with a sh and my commandline with a bash shell. Could this be a solution?

Now I think I must wait for next sunday an we will see, if the cronjob with shell=/bin/bash, runs again.

**Comment by annando** [comment]
Which version are you using? The RC should now be fine.

**Comment by gitkald** [comment]
Master
